test/community/assortativity.jl

```diff
@@ -20,4 +20,28 @@ using Random, Statistics
         y = [indegree(g, dst(d))-1 for d in edges(g)]
         @test @inferred assort ≈ cor([x;y], [y;x])
     end
+
+    # Test definition of assortativity as Pearson correlation coefficient
+    # between some scalar attributes characterising each vertex
+    @testset "Small graphs with attributes" for n = 5:2:11
+        g = grid([1,n])
+        attributes = [(-1.)^v for v in vertices(g)]
```

Comment:
it's not a good idea to assume `vertices` is an integer that you can use (it is for now, but not a great idea to rely on this behavior). Use `1:nv(g)` for times when you want a numeric value.